lib/common/common/src/types.rs

```diff
@@ -26,3 +26,35 @@ impl PartialOrd for ScoredPointOffset {
         Some(self.cmp(other))
     }
 }
+
+#[derive(Copy, Clone, Debug)]
+pub struct TelemetryDetail {
+    pub level: DetailsLevel,
+    pub histograms: bool,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum DetailsLevel {
+    Level0,
+    Level1,
+    Level2,
+}
+
+impl Default for TelemetryDetail {
+    fn default() -> Self {
+        TelemetryDetail {
+            level: DetailsLevel::Level0,
+            histograms: false,
+        }
+    }
+}
+
+impl From<usize> for DetailsLevel {
+    fn from(value: usize) -> Self {
+        match value {
+            0 => DetailsLevel::Level0,
+            1 => DetailsLevel::Level1,
+            _ => DetailsLevel::Level2,
+        }
+    }
+}
```

Comment:
We can derive `Default` like this, without any custom implementation:

```rust
#[derive(Copy, Clone, Debug, Default)]
pub struct TelemetryDetail {
    pub level: DetailsLevel,
    pub histograms: bool,
}

#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
pub enum DetailsLevel {
    #[default]
    Level0,
    Level1,
    Level2,
}
```

What do you think? I don't have a strong preference.

Reply:
I'd prefer not to have `Default` at all in the production code so it'll be always specified explicitly at the top-level.
However, it's convenient for unit/integration tests where concrete values are not important.

So, as a compromise I'd prefer to keep `Default` for `TelemetryDetail` (used in a test), but not for `DetailsLevel`.

Reply:
Check! But the we probably want to add `#[cfg(test)]` to the `impl Default` we have in this PR.

Reply:
Items under `#[cfg(test)]` couldn't be used in integration tests as they're separate crates. :( Only unit tests could use them.